db/migrate/20170529142557_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,169 @@
+         run_as_user run_as_non_root capabilities_add capabilities_drop command).each do |column|
+        join_sql = definitions.project(definitions[column.to_sym])
+                              .where(definitions[:id].eq(containers[:container_definition_id])).to_sql
+        Container.update_all("#{column} = (#{join_sql})")
+      end
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      port_configs = Arel::Table.new(:container_port_configs)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(port_configs[:container_definition_id])).to_sql
+      ContainerPortConfig.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      env_vars = Arel::Table.new(:container_env_vars)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(env_vars[:container_definition_id])).to_sql
+      ContainerEnvVar.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for security_contexts") do
+      security_contexts = Arel::Table.new(:security_contexts)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(security_contexts[:resource_id])
+                           .and(security_contexts[:resource_type].eq(Arel::Nodes::Quoted.new('ContainerDefinition')))).to_sql
+      SecurityContext.where(:resource_type => 'ContainerDefinition').update_all("resource_type = 'Container', resource_id = (#{join_sql})")
+    end
+
+    # relationships
+    rename_column :container_port_configs, :container_definition_id, :container_id
+    rename_column :container_env_vars, :container_definition_id, :container_id
+
+    remove_column :containers, :container_definition_id
+    drop_table :container_definitions
+  end
+
+  def down
+    create_table :container_definitions do |t|
+      t.belongs_to :ems, :type => :bigint
+      t.string     :ems_ref
+      t.bigint     :old_ems_id
+      t.timestamp  :deleted_on
+      t.string     :name
+      t.string     :image
+      t.string     :image_pull_policy
+      t.string     :memory
+      t.float      :cpu_cores
+      t.belongs_to :container_group, :type => :bigint
+      t.boolean    :privileged
+      t.bigint     :run_as_user
+      t.boolean    :run_as_non_root
+      t.string     :capabilities_add
+      t.string     :capabilities_drop
+      t.text       :command
+    end
+
+    add_column :containers, :container_definition_id, :bigint
+
+    say_with_time("splitting columns from container into container_definition") do
+      ContainerDefinition.transaction do
+        Container.all.each do |container|
```

Comment:
maybe temporary column `ContainerDefinition#container_id` would make sense.

That way, this could be done in 2 queries.